libs/dmlf/include/dmlf/queue.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include <memory>
+#include <queue>
+
+#include "core/mutex.hpp"
+
+#include "dmlf/queue_interface.hpp"
+#include "dmlf/update.hpp"
+#include "dmlf/update_interface.hpp"
+
+namespace fetch {
+namespace dmlf {
+
+template <typename T>
+class Queue : public QueueInterface
```

Comment:
Priority is actually fairly helpful in this context. That's because these queues will presumably hold training 'Updates'. As a general rule, older Updates will be less useful; thus a controller might choose to ignore them or apply them based on their position in the queue.